interactions/api/events/processors/voice_events.py

```diff
@@ -13,15 +13,20 @@
 class VoiceEvents(EventMixinTemplate):
     @Processor.define()
     async def _on_raw_voice_state_update(self, event: "RawGatewayEvent") -> None:
-        before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
         after = await self.cache.place_voice_state_data(event.data)
+        is_bot = event.data["member"]["user"]["bot"]
 
-        self.dispatch(events.VoiceStateUpdate(before, after))
-
-        if before and before.user_id == self.user.id:
-            if vc := self.cache.get_bot_voice_state(event.data["guild_id"]):
+        if is_bot:
+            before_bot_guild_voice_state = self.cache.get_bot_voice_state(event.data["guild_id"])
+            self.dispatch(events.VoiceStateUpdate(before_bot_guild_voice_state, after))
+            if vc := before_bot_guild_voice_state:
                 # noinspection PyProtectedMember
-                await vc._voice_state_update(before, after, event.data)
+                await vc._voice_state_update(after, event.data)
+            return
+
+        # Following code is for updates to non-bot users
+        before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
+        self.dispatch(events.VoiceStateUpdate(before, after))
```

Comment:
This logic seems a bit redundant - I'm pretty sure you can do something like:

```python
if is_bot:
    before = self.cache.get_bot_voice_state(event.data["guild_id"])
    if vc := before:
        # noinspection PyProtectedMember
        await vc._voice_state_update(after, event.data)
else:
    # Following code is for updates to non-bot users
    before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None

self.dispatch(events.VoiceStateUpdate(before, after))
```

This would also make sure the other events below it get dispatched correctly.

Reply:
👍